spec/controllers/idv/hybrid_mobile/socure/document_capture_controller_spec.rb

```diff
@@ -0,0 +1,166 @@
+require 'rails_helper'
+
+RSpec.describe Idv::HybridMobile::Socure::DocumentCaptureController do
+  include FlowPolicyHelper
+
+  let(:idv_vendor) { Idp::Constants::Vendors::SOCURE }
+  let(:fake_socure_endpoint) { 'https://fake-socure.com' }
+  let(:user) { create(:user) }
+  let(:stored_result) { nil }
+
+  let(:document_capture_session) do
+    DocumentCaptureSession.create(
+      user: user,
+      requested_at: Time.zone.now,
+    )
+  end
+  let(:document_capture_session_uuid) { document_capture_session&.uuid }
+
+  before do
+    allow(IdentityConfig.store).to receive(:socure_document_request_endpoint).
+      and_return(fake_socure_endpoint)
+    allow(IdentityConfig.store).to receive(:doc_auth_vendor).and_return(idv_vendor)
+    allow(IdentityConfig.store).to receive(:doc_auth_vendor_default).and_return(idv_vendor)
+
+    allow(subject).to receive(:stored_result).and_return(stored_result)
+  end
+
+  describe 'before_actions' do
+    it 'checks valid document capture session' do
+      expect(subject).to have_actions(
+        :before,
+        :check_valid_document_capture_session,
+      )
+    end
+  end
+
+  describe '#show' do
+    let(:request_class) { DocAuth::Socure::Requests::DocumentRequest }
+
+    let(:expected_language) { :en }
+    let(:response_body) { {} }
+
+    before do
+      stub_request(:post, fake_socure_endpoint).to_return(
+        status: 200,
+        body: JSON.generate(response_body),
+      )
+
+      session[:doc_capture_user_id] = user&.id
+      session[:document_capture_session_uuid] = document_capture_session_uuid
+    end
+
+    context 'with no user id in session' do
+      let(:document_capture_session) { nil }
+      let(:user) { nil }
+
+      it 'redirects to root' do
+        get :show
+        expect(response).to redirect_to root_url
+      end
+    end
+
+    context 'happy path' do
+      let(:response_redirect_url) { 'https://boogie-woogie.com/dance' }
```

Comment:
While I don't mind a bit of humor in our tests, I am a bit concerned about pointing to domain names that can potentially contain inappropriate or malicious content.

I'd at least consider using the [`.test` top-level domain, which is guaranteed to never be registered](https://en.wikipedia.org/wiki/.test).

```suggestion
      let(:response_redirect_url) { 'https://boogie-woogie.test/dance' }
```

Reply:
Changed to test.org. Path is still /dance.

Reply:
`test.org` is actually [a registered domain](https://www.whois.com/whois/test.org), although there doesn't currently appear to be an IP associated with it.

Reply:
Changed to `idv.test`